ovs/dal/datalist.py

```diff
@@ -203,7 +203,13 @@ def _execute_query(self):
         prefix = '{0}_{1}_'.format(DataObject.NAMESPACE, object_type_name)
 
         if self._guids is not None:
-            entries = list(self._persistent.get_multi(['{0}{1}'.format(prefix, guid) for guid in self._guids]))
+            try:
+                entries = list(self._persistent.get_multi(['{0}{1}'.format(prefix, guid) for guid in self._guids]))
```

Comment:
Can you build a list of the keys first? Something like:

```
keys = ['{0}{1}'.format(prefix, guid) for guid in self._guids]
```

Then, if an error occurs, you can remove the key immediately from that list, and then parse out the guid to remove it from _guids. This will make sure you don't have loop through all guids every time you retry.

Also, like stated in the previous comment, please re-try this 5 times (analog code like in ```DataObject.delete()```. And it might be good to in that case raise a ```RaceConditionException```, just like the delete mentioned earlier.